### DISCUSSION TITLE
Google sheets vs mysql database

### DISCUSSION THREAD

**Discussion by Paulmski** [Paulmski/Discord-Bot#20] [Ideas]
What are people's thoughts on using Google sheets vs a MySQL database? I could see how a spread sheet could be easier upfront, but could we encounter challenges as the project progresses and more information is stored. On the other hand, a database could be cumbersome to first create and design but we might benefit from it in the long run. I'm not an expert in either of these areas so others' input would be much appreciated.

**Comment by tuanyvan** [comment]
The good thing about Google Sheets is that it's easy to write to, especially as the end user. However, Google's API is [very difficult to work with](https://developers.google.com/workspace/guides/create-credentials) and get OAuth 2.0 Client ID, credentials.json, etc., making reading from Sheets not easy to do. It makes it especially difficult for future contributors to run the code on their own.

Looking at [the MySQL documentation](https://dev.mysql.com/doc/connector-python/en/connector-python-example-connecting.html), reading to the database is ridiculously easy (no need to make OAuth 2.0 IDs and export to credentials.json). Writing to the database would be relatively easy if we had a Python script for pushing/popping values.

All in all, I agree with using a MySQL database rather than Google Sheets because of how difficult it is to link the Google API to the Discord-Bot.

Edit: I got finally got the Google API to work shortly after this reply, but it is still undeniable that the process of getting the credentials.json file from the Google Cloud Platform would not be easy for everyone.
